x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/interceptor/SearchRequestInterceptor.java

```diff
@@ -69,7 +69,11 @@ void disableFeatures(
 
     @Override
     public boolean supports(IndicesRequest request) {
-        return request instanceof SearchRequest;
+        if (request instanceof SearchRequest searchRequest) {
+            return searchRequest.source() != null && (searchRequest.source().suggest() != null || searchRequest.source().profile());
+        } else {
+            return false;
+        }
```

Comment:
This is a problem.
The interceptor also disables the request cache:
```
        final SearchRequest request = (SearchRequest) indicesRequest;
        if (hasRemoteIndices(request)) {
            request.requestCache(false);
        }
```

Reply:
Good catch. I introduced a new interceptor (`SearchRequestCacheDisablingInterceptor`) which only disables the cache if the request has remote indices.

Reply:
LGTM!

Minor additional nit: can you please extract the two conditions `source != null && source.suggest() != null` and `source != null && source.profile()` in two private methods that you then reuse in the `supports` and `disableFeatures` methods?